File: MPP/Laborator1/src/main/java/service/ServiceAdmin.java
package service;

import model.TipUser;
import model.User;
import repository.IRepositoryUser;
import utils.DataChanged;
import utils.Observable;
import utils.Observer;
import utils.PasswordStorage;

import java.util.ArrayList;
import java.util.List;

import static utils.PasswordStorage.createHash;

//adminul adauga probe, operatori
public class ServiceAdmin implements Observable<DataChanged> {
    private List<Observer<DataChanged>> observers=new ArrayList<>();
    private IRepositoryUser repository;

    public ServiceAdmin(IRepositoryUser repository) {
        this.repository = repository;
    }

    public User createUser(String username, String parola) throws PasswordStorage.CannotPerformOperationException {
        String hash= createHash(parola);
        User user=new User(username,hash, TipUser.OPERATOR);
        repository.save(user);
        return user;
    }

    public User createUser() throws PasswordStorage.CannotPerformOperationException {
        String hash= createHash("admin");
        User user=new User("admin",hash, TipUser.ADMIN);
        repository.save(user);
        return user;
    }

    public boolean verificareParola(String username, String password) {
        User user = repository.findOne(username);

        if (user!=null) {
            String userHash = user.getHash();
            try {
                return PasswordStorage.verifyPassword(password, userHash);
            } catch (PasswordStorage.CannotPerformOperationException e) {
                e.printStackTrace();
            } catch (PasswordStorage.InvalidHashException e) {
                e.printStackTrace();
            }
        }
        return false;
    }

    public User cauta(String username){
        User entity= repository.findOne(username);
        return entity;
    }


    @Override
    public void addObserver(Observer<DataChanged> e) {
        observers.add(e);
    }

    @Override
    public void removeObserver(Observer<DataChanged> e) {
        observers.remove(e);
    }

    @Override
    public void notifyObservers(DataChanged t) {
        observers.forEach(o->o.update(t));
    }
}

File: MPP/LabC#/WindowsFormsApp1/service/ServiceAdmin.cs
using Concurs.model;
using Concurs.repository;
using Concurs.utils;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Concurs.service
{
    class ServiceAdmin: Subject
    {
        private IRepositoryUser repository;

        public ServiceAdmin(IRepositoryUser repository)
        {
            this.repository = repository;
        }

        public User CreateUser(string username, string parola)
        {  
            string hash= PasswordStorage.CreateHash(parola);
            User user = new User(username, hash, "OPERATOR");
            repository.Save(user);
            return user;
        }

        public User CreateUser() 
        {
            string hash= PasswordStorage.CreateHash("admin");
            User user=new User("admin", hash, "ADMIN");
            repository.Save(user);
            return user;
        }

        public bool VerificareParola(string username, string password)
        {
            User user = repository.FindOne(username);

            if (user != null)
            {
                string userHash = user.Hash;
                if (userHash == PasswordStorage.CreateHash(password))
                {
                    return true;
                }             
            }
            return false;
        }

        public User Cauta(string username)
        {
            User entity = repository.FindOne(username);
            return entity;
        }
    }
}
